opentelemetry-sdk/src/logs/record.rs

```diff
@@ -37,6 +39,22 @@ pub struct LogRecord {
     pub attributes: Option<Vec<(Key, AnyValue)>>,
 }
 
+impl Default for LogRecord {
+    fn default() -> Self {
+        LogRecord {
+            event_name: None,
+            target: None,
+            timestamp: None,
+            observed_timestamp: None,
+            trace_context: None,
+            severity_text: None,
+            severity_number: None,
+            body: None,
+            attributes: Some(Vec::with_capacity(PREALLOCATED_ATTRIBUTE_CAPACITY)), // Pre-allocate for perf optimization. This may change in future.
```

Comment:
question: Should we micro-optimize this to only allocate attributes, when first attribute is added, so that if someone is simply using body only, and no attributes, then we can save some space...

Reply:
That's a good point. Also to add, as of now we are always adding `meta.name` to [attributes](https://github.com/open-telemetry/opentelemetry-rust/blob/ad990d6d33aa73b35e0e3b792b1fe3ea4251b92b/opentelemetry-appender-tracing/src/layer.rs#L43), so we won't get benefit of above optimization. The event-name should be instead populated as  the struct member `LogRecord::name` (after #1488). Can incorporate both these changes as separate PR.